integration-tests/capacity/transactions.test.ts

```diff
@@ -22,555 +22,604 @@ import {
-            it("fails to pay for transaction when key does has not met the min deposit", async function () {
-                const noTokensKeys = createKeys("noTokensKeys");
-                const delegatorKeys = await createAndFundKeypair(2n * DOLLARS, "delegatorKeys");
-
-                await assert.doesNotReject(stakeToProvider(capacityKeys, capacityProvider, 1n * DOLLARS));
-
-                // Add new key
-                let newKeyPayload: AddKeyData = await generateAddKeyPayload({
-                     msaId: new u64(ExtrinsicHelper.api.registry, capacityProvider), newPublicKey: noTokensKeys.publicKey
-                });
-                let addKeyData = ExtrinsicHelper.api.registry.createType("PalletMsaAddKeyData", newKeyPayload);
-
-                let ownerSig = signPayloadSr25519(capacityKeys, addKeyData);
-                let newSig = signPayloadSr25519(noTokensKeys, addKeyData);
-                const addPublicKeyOp = ExtrinsicHelper.addPublicKeyToMsa(capacityKeys, ownerSig, newSig, newKeyPayload);
-
-                const [publicKeyEvents] = await addPublicKeyOp.fundAndSend();
-                assert.notEqual(publicKeyEvents, undefined, 'should have added public key');
-
-                const createMsaOp = ExtrinsicHelper.createMsa(delegatorKeys);
-                const [MsaCreatedEvent] = await createMsaOp.fundAndSend();
-                assert.notEqual(MsaCreatedEvent, undefined, "should have returned MsaCreated event");
-
-                const payload = await generateDelegationPayload({
-                    authorizedMsaId: capacityProvider,
-                    schemaIds: [schemaId],
-                });
-                const addProviderData = ExtrinsicHelper.api.registry.createType("PalletMsaAddProvider", payload);
-                const grantDelegationOp = ExtrinsicHelper.grantDelegation(delegatorKeys, noTokensKeys,
-                    signPayloadSr25519(delegatorKeys, addProviderData), payload);
-
-                await assert.rejects(grantDelegationOp.payWithCapacity(), {
-                    name: 'RpcError',
-                    message: /Custom error: 3/,
-                })
-            });
+  describe("pay_with_capacity", function () {
+    describe("when caller has a Capacity account", async function () {
+      let capacityKeys: KeyringPair;
+      let capacityProvider: u64;
+      let schemaId: u16;
+      const amountStaked = 2n * DOLLARS;
+
+      before(async function () {
+        capacityKeys = createKeys("CapacityKeys");
+        capacityProvider = await createMsaAndProvider(capacityKeys, "CapacityProvider", FUNDS_AMOUNT);
+
+        // Create schemas for testing with Grant Delegation to test pay_with_capacity
+        schemaId = await getOrCreateGraphChangeSchema();
+        assert.notEqual(schemaId, undefined, "setup should populate schemaId");
+      });
+
+      async function assertAddNewKey(addKeyPayload: AddKeyData, newControlKeypair: KeyringPair) {
```

Comment:
The formatter kind of messed things up for the diffs.  If you hide the whitespace it'll be easier to read.